File: gyroTest/src/main/java/com/stulnikov/arduino/arduinoBt/gyrotest/gyrotest/sensor/AccelerometerProvider.java
package com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.sensor;

/**
 * @author alexeystulnikov 12/28/16.
 */

public interface AccelerometerProvider {

    interface SensorCallback {
        void onSensorXChanged(float x);
        void onSensorYChanged(float y);
        void onSensorZChanged(float z);
        void onAccuracyChanged(int accuracy);
    }

    void registerListener(SensorCallback callback);
    void unregisterListener(SensorCallback callback);
}

File: gyroTest/src/main/kotlin/com/stulnikov/arduino/arduinoBt/gyrotest/gyrotest/sensor/AccelerometerProvider.kt
package com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.sensor

/**
 * @author alexeystulnikov 12/28/16.
 */
interface AccelerometerProvider {
    interface SensorCallback {
        fun onSensorXChanged(x: Float)
        fun onSensorYChanged(y: Float)
        fun onSensorZChanged(z: Float)
        fun onAccuracyChanged(accuracy: Int)
    }

    fun registerListener(callback: SensorCallback?)
    fun unregisterListener(callback: SensorCallback?)
}